sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py

```diff
@@ -66,7 +66,10 @@ def get_length(data):
         except (AttributeError, UnsupportedOperation):
             pass
         else:
-            return fstat(fileno).st_size
+            try:
```

Comment:
a comment here could be perfect!!(explaining why we need to try inside of else block)

Reply:
done